In [64]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
import joblib
import pickle

# Specific Imports
from sklearn.datasets import load_boston
from sklearn.linear_model import LogisticRegression
from sklearn import metrics
from sklearn.metrics import mean_squared_error
from sklearn.model_selection import train_test_split, KFold, cross_val_score

In [65]:
X_test = pd.read_csv("data/prepared/X_test_ScoreNewData.csv")

In [66]:
y_test = pd.read_csv("data/prepared/y_test_ScoreNewData.csv")

# Load Random Forest Model

In [15]:
from joblib import dump, load#

# Load fthe model from the file
rfc_from_joblib = load('data/prepared/rfc_mfrancis870.pkl')

In [16]:
result = rfc_from_joblib.score(X_test, y_test)
print(result)

0.854399243140965


In [17]:
#data format 
X_test.columns   # input Data format for x

Index(['Income', 'GenHlth', 'MentHlth', 'PhysHlth', 'DiffWalk'], dtype='object')

In [52]:
#creat new dataframe for new data
new_data_df = pd.DataFrame()

In [53]:
#add columns to new dataframe
new_data_X = pd.DataFrame(new_data_df, columns=['Income', 'GenHlth', 'MentHlth', 'PhysHlth', 'DiffWalk'])

In [55]:
new_data_X.loc[0] = [2,3,1,0,0] 
new_data_X 

Income  GenHlth  MentHlth  PhysHlth  DiffWalk
0       2        3         1         0         0

In [63]:
y_pred = rfc_from_joblib.predict(new_data_X ) 
print(y_pred)

[0]





Given income Less than $15,000, good general health, no days where health was not good, and no diffi walking , you have a 85% proababliy you will not be diagnosed with   diabetes